src/c++/perf_analyzer/genai-perf/genai_perf/llm_inputs/llm_inputs.py

```diff
@@ -106,83 +69,40 @@ def create_llm_inputs(
-        dataset_name:
-            The name of the dataset
-        model_name:
-            The model name
-        starting_index:
-            Offset from within the list to start gathering inputs
-        length:
-            Number of entries to gather
-        add_model_name:
-            If true, adds a model name field to each payload
-        add_stream:
-            If true, adds a steam field to each payload
-        extra_inputs:
-            If provided, append these inputs to every request
-        output_tokens_mean:
-            The mean length of the output to generate. If not using fixed output lengths, this should be set to -1.
-        output_tokens_stddev:
-            The standard deviation of the length of the output to generate. This is only used if output_tokens_mean is provided.
-        output_tokens_deterministic:
-            If true, the output tokens will set the minimum and maximum tokens to be equivalent.
-        batch_size:
-            The number of inputs per request (currently only used for the embeddings and rankings endpoints)
-
-        Required Synthetic Prompt Generation Parameters
-        -----------------------------------------------
-        tokenizer:
-           The tokenizer to use when generating synthetic prompts
-
-        Optional Synthetic Prompt Generation Parameters
-        -----------------------------------------------
-        prompt_tokens_mean:
-            The mean length of the prompt to generate
-        prompt_tokens_stddev:
-            The standard deviation of the length of the prompt to generate
-        num_of_output_prompts:
-            The number of synthetic output prompts to generate
-        random_seed:
-            Seed used to generate random values
-        """
-
-        cls._check_for_valid_args(
-            input_type, dataset_name, starting_index, length, tokenizer
+        cls.validate_args(
+            input_type, output_format, dataset_name, starting_index, length, tokenizer
         )
 
         random.seed(random_seed)
 
-        generic_dataset_json = cls.get_generic_dataset_json(
-            input_type,
-            output_format,
-            dataset_name,
-            starting_index,
-            length,
-            tokenizer,
-            prompt_tokens_mean,
-            prompt_tokens_stddev,
-            num_of_output_prompts,
-            batch_size,
-            input_filename,
-        )
+        if input_type == PromptSource.DATASET:
+            dataset = DatasetRetriever.from_url(
+                cls.dataset_url_map[dataset_name], starting_index, length
+            )
+        elif input_type == PromptSource.SYNTHETIC:
+            dataset = DatasetRetriever.from_synthetic(
+                tokenizer,
+                prompt_tokens_mean,
+                prompt_tokens_stddev,
+                num_of_output_prompts,
+            )
+        elif input_type == PromptSource.FILE:
+            input_filename = cast(Path, input_filename)
```

Comment:
duplicate in line 95?

Reply:
It might be necessary for the type checking to pass but worth double checking.